### PR TITLE
Basic Metrics for KCBQ

### PR DESCRIPTION
**Pull request by mtagle**
metrics for the average, max, and rate for data reading from kafka and data writing to bq.

### CODE REVIEW

File: src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java [hunk 1]
```diff
@@ -286,12 +295,30 @@ private BigQueryWriter getWriter() {
...
   }
 
+  private void configureMetrics() {
+    // todo more metric configuration?
+    metrics = new Metrics();
+    rowsRead = metrics.sensor("rows-read");
+    rowsRead.add(metrics.metricName("rows-written-avg",
+                                    "kafka", // I have no idea how the group name is used or what it should be...
+                                    "The average number of rows written per request"),
+                 new Avg());
```
Comment (criccomini) [inline src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java]: Nit: IntelliJ formatter is being annoying. This should be indented to match the line above, or the `);` should be moved to a newline and `new Avg()` should be properly indented.

</ocd>

File: src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java [hunk 2]
```diff
@@ -286,12 +295,30 @@ private BigQueryWriter getWriter() {
...
+      return new SimpleBigQueryWriter(bigQuery, retry, retryWait, metrics);
     }
   }
 
+  private void configureMetrics() {
+    // todo more metric configuration?
+    metrics = new Metrics();
+    rowsRead = metrics.sensor("rows-read");
+    rowsRead.add(metrics.metricName("rows-written-avg",
+                                    "kafka", // I have no idea how the group name is used or what it should be...
```
Comment (criccomini) [inline src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java]: Use getClass().toString() as the group name.

File: src/main/java/com/wepay/kafka/connect/bigquery/write/BigQueryWriter.java [hunk 3]
```diff
@@ -40,12 +45,28 @@
...
   private long retryWaitMs;
 
-  public BigQueryWriter(int retries, long retryWaitMs) {
+  public BigQueryWriter(int retries, long retryWaitMs, Metrics metrics) {
     this.retries = retries;
     this.retryWaitMs = retryWaitMs;
+
+    rowsWritten = metrics.sensor("rows-written");
+    rowsWritten.add(metrics.metricName("rows-written-avg",
+                                       "big-query", // I have no idea how the group name is used or what it should be...
```
Comment (criccomini) [inline src/main/java/com/wepay/kafka/connect/bigquery/write/BigQueryWriter.java]: getClass().getCanonicalName()

**Comment by criccomini** [comment]
This looks good. Couple of minor comments.

File: src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java (hunk 1 above)
Comment (mtagle) [inline src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java]: new Avg() seems like it is properly indented to me.  it's an argument of rowsRead.add(...)

File: src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java (hunk 2 above)
Comment (mtagle) [inline src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java]: You really want the source class to be the group name of the metric?  That isn't what seems to be happening in kafka proper.  I was actually going to rename the group as "kcbq" or something.

File: src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java (hunk 1 above)
Comment (criccomini) [inline src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java]: NM, I misread this line.

File: src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java (hunk 2 above)
Comment (criccomini) [inline src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java]: Hmm, I could be confused about what group name means. I'll defer to whatever pattern Kafka is using.

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/BigQueryWriter.java [hunk 4]
```diff
@@ -43,12 +49,28 @@
...
   private long retryWaitMs;
 
-  public BigQueryWriter(int retries, long retryWaitMs) {
+  public BigQueryWriter(int retries, long retryWaitMs, Metrics metrics) {
     this.retries = retries;
     this.retryWaitMs = retryWaitMs;
+
+    rowsWritten = metrics.sensor("rows-written");
+    rowsWritten.add(metrics.metricName("rows-written-avg",
+                                       "kcbq",
```
Comment (criccomini) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/BigQueryWriter.java]: Seems like this be a constant or something

**Comment by criccomini** [comment]
LGTM just a minor nit

**Pull request merged**